### PR TITLE
Add try-catch statement to the body of generated main method

### LINKED ISSUE #21575: Java stacktrace printed when a metric extension fails to initialize
**Description:**

When a tracing or a metrics extension fails in the `init` method a Java stacktrace is shown to the user. It is better if we can show a ballerina error since this is a user error.

```
Exception in thread "main" error Choreo client is not initialized
        at org.ballerinalang.jvm.values.api.BValueCreator.createErrorValue(BValueCreator.java:217)
        at org.ballerinalang.observe.trace.extension.choreo.MetricsReporterExtension.init(MetricsReporterExtension.java:61)
        at org.ballerinalang.jvm.observability.metrics.MetricsLaunchListener.beforeRunProgram(MetricsLaunchListener.java:77)
        at org.ballerinalang.jvm.launch.LaunchUtils.lambda$startListeners$0(LaunchUtils.java:98)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.ballerinalang.jvm.launch.LaunchUtils.startListeners(LaunchUtils.java:98)
        at choreo.hello-service.___init.main(hello-service)
```

**Steps to reproduce:**

1. Run a Ballerina program with incorrect Choreo tracing extension configs.

`ballerina.conf`
```
[b7a.observability.tracing]
enabled=true
name="choreo"

[b7a.observability.metrics]
enabled=true
reporter="choreo"

[b7a.observability.tracing.choreo.reporter]
hostname="foo.bar"
port="443"
```

**Affected Versions:**

1.2.0-SNAPSHOT

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Kishanthan**
## Purpose
This PR adds a common try-catch to generated java main method. This will now handle any runtime exceptions or errors thrown from program initialization and shutdown (for example: launch listener) and will handle them in a graceful manner so that user sees proper ballerina related errors on the console. 

Fixes #21575

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java [hunk 1]
```diff
@@ -1722,7 +1727,7 @@ static void generateMainMethod(@Nilable BIRFunction userMainFunc, ClassWriter cw
             mv.visitIntInsn(BIPUSH, 100);
             mv.visitTypeInsn(ANEWARRAY, OBJECT);
             mv.visitFieldInsn(PUTFIELD, STRAND, "frames", String.format("[L%s;", OBJECT));
-            errorGen.printStackTraceFromFutureValue(mv, indexMap);
+            handleErrorFromFutureValue(mv);
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java:1730]: can we remove the original `handleErrorFromFutureValue` method?

File: tests/launch-listener-test-util-libs/test-launch-listener-03/spotbugs-exclude.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<!--
+  ~  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (warunalakshitha) [inline tests/launch-listener-test-util-libs/test-launch-listener-03/spotbugs-exclude.xml:2]: Do we need this empty spotbugs-exclude files

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java [hunk 3]
```diff
@@ -1685,7 +1690,7 @@ static void generateMainMethod(@Nilable BIRFunction userMainFunc, ClassWriter cw
             mv.visitIntInsn(BIPUSH, 100);
             mv.visitTypeInsn(ANEWARRAY, OBJECT);
             mv.visitFieldInsn(PUTFIELD, STRAND, "frames", String.format("[L%s;", OBJECT));
-            errorGen.printStackTraceFromFutureValue(mv, indexMap);
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java:1688]: can remove the original method ?

Comment (Kishanthan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java:1688]: yes, done

File: tests/launch-listener-test-util-libs/test-launch-listener-03/spotbugs-exclude.xml (hunk 2 above)
Comment (Kishanthan) [reply tests/launch-listener-test-util-libs/test-launch-listener-03/spotbugs-exclude.xml]: no need. removed.

**Pull request merged**